src/panels/lovelace/cards/hui-entity-card.ts

```diff
@@ -0,0 +1,270 @@
+
+  public setConfig(config: EntityCardConfig): void {
+    if (config.entity && !isValidEntityId(config.entity)) {
+      throw new Error("Invalid Entity");
+    }
+
+    this._config = config;
+  }
+
+  public getCardSize(): number {
+    return 2;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+
+    if (!stateObj) {
+      return html`
+        <hui-warning
+          >${this.hass.localize(
+            "ui.panel.lovelace.warning.entity_not_found",
+            "entity",
+            this._config.entity
+          )}</hui-warning
+        >
+      `;
+    }
+
+    const showUnit =
+      (!this._config.attribute &&
+        stateObj.state !== UNKNOWN &&
+        stateObj.state !== UNAVAILABLE) ||
+      (this._config.attribute && stateObj.attributes[this._config.attribute]);
+
+    return html`
+      <ha-card
+        @action=${this._handleClick}
+        .actionHandler=${actionHandler()}
+        tabindex="0"
+      >
+        <div class="flex header">
+          <div class="name">
+            <span>${this._config.name || computeStateName(stateObj)}</span>
+          </div>
+          <div class="icon">
+            <ha-icon
+              .icon=${this._config.icon || stateIcon(stateObj)}
+            ></ha-icon>
+          </div>
+        </div>
+        <div class="flex info">
+          <span id="value"
+            >${this._config.attribute
+              ? stateObj.attributes[this._config.attribute] ||
+                this.hass.localize("state.default.unknown")
+              : stateObj.state}</span
+          >${showUnit
+            ? html`
+                <span id="measurement"
+                  >${this._config.unit ||
+                    stateObj.attributes.unit_of_measurement}</span
+                >
+              `
+            : ""}
+        </div>
+        ${this._config.footer
+          ? this.renderHeaderFooter(this._config.footer, "footer")
+          : ""}
+      </ha-card>
+    `;
+  }
+
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    return hasConfigOrEntityChanged(this, changedProps);
+  }
+
+  protected updated(changedProps: PropertyValues) {
+    super.updated(changedProps);
+    if (!this._config || !this.hass) {
+      return;
+    }
+
+    const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
+    const oldConfig = changedProps.get("_config") as
+      | EntityCardConfig
+      | undefined;
+
+    if (
+      !oldHass ||
+      !oldConfig ||
+      oldHass.themes !== this.hass.themes ||
+      oldConfig.theme !== this._config.theme
+    ) {
+      applyThemesOnElement(this, this.hass.themes, this._config!.theme);
+    }
+  }
+
+  private renderHeaderFooter(
+    conf: LovelaceHeaderFooterConfig,
+    className: string
+  ): TemplateResult {
+    const element = createHeaderFooterElement(conf);
```

Comment:
Now we're throwing away and re-creating this element at every render. And since `hass` is a property, that's a lot. You should create this element in `setConfig` and update it in `updated` with new `hass`

Reply:
We should probably update the entities card then, no?

Reply:
Nope, the entities card has a setter for `hass`. It only re-renders when new config arrives, otherwise it manually looks at the DOM elements and finds the ones that need `hass` updated.

Reply:
You cannot do that here because you are rendering a template based on hass-data. Entities card just needs to pass it along to the rows/header/footer

Reply:
Got it. Makes sense